### PR TITLE
chore: document dependency injection

### PR DESCRIPTION
**Pull request by paul-vd**
## What?
This documents:
- Dependency Injection
- Services
- Dynamic pages sitemap registration
- Migration Guide

## Preview
- [/guides/dependency-injection](https://deploy-preview-871--heuristic-almeida-1a1f35.netlify.app/docs/3.x/guides/dependency-injection)
- [/guides/customize-the-sitemap](https://deploy-preview-871--heuristic-almeida-1a1f35.netlify.app/docs/3.x/guides/customize-the-sitemap)
- [/api-reference/front-commerce-core/createSitemapFetcher](https://deploy-preview-871--heuristic-almeida-1a1f35.netlify.app/docs/3.x/api-reference/front-commerce-core/createSitemapFetcher)
- [/api-reference/front-commerce-core/services](https://deploy-preview-871--heuristic-almeida-1a1f35.netlify.app/docs/3.x/api-reference/front-commerce-core/services)
- [/migration/manual-migration#sitemap-changes](https://deploy-preview-871--heuristic-almeida-1a1f35.netlify.app/docs/3.x/migration/manual-migration#sitemap-changes)

### CODE REVIEW

File: docs/02-guides/dependency-injection.mdx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+```ts
+import { json } from "@front-commerce/remix/node";
+import type { LoaderFunctionArgs } from "@remix-run/node";
+
+export const loader = async ({ context }: LoaderFunctionArgs) => {
+  const app = new FrontCommerceApp(context.frontCommerce);
+
+  // highlight-start
+  const { client } = services.DI.acme;
```
Comment (mGGk-fr) [inline docs/02-guides/dependency-injection.mdx:105]: `app.services.DI.acme`

File: docs/02-guides/dependency-injection.mdx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+```ts
+import { json } from "@front-commerce/remix/node";
+import type { ActionFunctionArgs } from "@remix-run/node";
+
+export const action = async ({ context }: ActionFunctionArgs) => {
+  const app = new FrontCommerceApp(context.frontCommerce);
+
+  // highlight-start
+  const { client } = services.DI.acme;
```
Comment (mGGk-fr) [inline docs/02-guides/dependency-injection.mdx:126]: `app.services.DI.acme`

**Review by mGGk-fr — APPROVED** [review]

**Review by Kagaminara — APPROVED** [review]

File: docs/02-guides/dependency-injection.mdx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+declare module "@front-commerce/types" {
+  export interface DependencyInjection {
+    acme: ReturnType<typeof createAcmeDI>;
+  }
+}
+```
+
+Make sure this types declaration file is included in your `tsconfig.json` file
+under the `include` or include sections to ensure TypeScript is aware of these
+custom types.
```
[the remark this answers is not in the document]
Comment (paul-vd) [reply docs/02-guides/dependency-injection.mdx:74]: I don't understand? diff for what? it's non existant so it's not diffing anywhere.

File: docs/02-guides/customize-the-sitemap.mdx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+:::
+
+## Dynamic Pages Customization
+
+Dynamic pages can also be customized using `getSitemapEntries` or a custom
+`sitemapFetcher`.
+
+### `getSitemapEntries` Function
+
+```tsx title="./extensions/acme/routes/todo.$id.tsx"
```
[the remark this answers is not in the document]
Comment (paul-vd) [reply docs/02-guides/customize-the-sitemap.mdx:57]: No, this is the `Dynamic Pages Customization` for the one you mentioned check the `Static Pages Customization`.

The parent route would never be responsable for listing entries of a dynamic, route.

To understand this concept, a good example would be with next.js Static Site Generation ([SSG](https://nextjs.org/docs/pages/building-your-application/rendering/static-site-generation)) [`getStaticPaths`](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-static-paths)

File: docs/02-guides/dependency-injection.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (paul-vd) [reply docs/02-guides/dependency-injection.mdx:74]: Ah for the `tsconfig.json`?

File: docs/02-guides/customize-the-sitemap.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Kagaminara) [reply docs/02-guides/customize-the-sitemap.mdx:57]: Oh you're right, indeed. Thanks !

File: docs/02-guides/customize-the-sitemap.mdx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+```tsx title="./extensions/acme/routes/todo.$slug.tsx"
+import { createSeoHandle } from "@front-commerce/remix/seo";
+
+export const handle = createSeoHandle({
+  sitemapFetcher: "todoSitemapFetcher",
+});
+```
+
+## Creating and Registering a Custom Fetcher
```
[the remark this answers is not in the document]
Comment (Kagaminara) [reply docs/02-guides/customize-the-sitemap.mdx:88]: That makes total sense, thank you very much.

File: docs/02-guides/dependency-injection.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Kagaminara) [reply docs/02-guides/dependency-injection.mdx:74]: > Ah for the tsconfig.json?

Yep 😃

**Pull request merged**